Extending use cases (#23)
temporalio/documentation-legacy: docs/use-cases-long-running.md

---
id: use-cases-long-running
title: Long-running Processes
sidebar_label: Long-running Processes
---

## Motivation

> Many business processes naturally have ф long duration and may run for hours, days, or months:

Reviewer: ф ?

- Expense approval process requiring manual intervention from a human.
- A process of labeling data for ML where an expert fills metadata via a user interface.
- Fraud detection system, where workflows react to events generated by consumer behavior.
- Customer loyalty program where the workflow accumulates reward points and applies them when requested.
- User nurturing process that sends educational emails based on the schedule and past user activities.

It's typical to use a distributed asynchronous event-driven architecture in such scenarios. The code is scattered across multiple handlers and does not resemble the structure of the original business process.

## Benefits of Temporal

Temporal provides a holistic approach to the implementation of long-running workflows. Workflow execution is paused and resumed by the runtime as required. Programmers don't have to focus on edge cases and boilerplate code of handling long periods of inactivity, and can concentrate on logic instead.

> Temporal has direct support for asynchronous events (aka signals). A simple programming model handles a lot of complexity around state persistence and ensures external action execution through built-in retries.

Author: `Temporal's`?

> A sophisticated workflow may have a few different paths that branch out, and you want to execute in steps parallel, but then join them back together, even if each track might take days to run. It's straightforward to express complicated decision paths because everything is represented as code rather than some GUI or JSON configuration.

Reviewer 2: ```suggestion
A sophisticated workflow may have a few different paths that branch out, that you want to execute in parallel, but then join them back together, even if each track might take days to run. With Temporal, it's straightforward to express complicated decision paths since everything is represented as code rather than via a GUI or JSON configuration.
```

Author: I tried avoiding double-`that`. Does it still read okay with double-that?
